content/en/events/upcoming-events/gsoc-2025.md

```diff
@@ -75,4 +75,306 @@ To participate in GSoC with Kubeflow, you __must__ meet the GSoC [eligibility re
 
 ## Project Ideas for 2025 GSoC
 
-Coming soon!
+### Project 1
+### Project Title: multi-tenancy/security, Kubeflow repository Migration, and CI/CD Enhancement
+
+#### Detailed Description:
+
+- Productionize the seaweedfs PoC as secure minio replacement https://github.com/kubeflow/manifests/tree/master/contrib/seaweedfs, since we are stuck with a 5 year old minio due to AGPL
+- add multi tenancy to seweadfs similar to the approach in feat(backend): isolate artifacts per namespace/profile/user using only one bucket pipelines#7725
+-Help to migrate Kubeflow/kubeflow components to kubeflow/dashboard and kubeflow/notebooks. This might even start before Gsoc
+- Renovate kubeflow/dashboard dependencies, addressing CVEs.
+- Implement architectural changes as needed.
+- Adapt and improve scripts and CI/CD in kubeflow/manifests, including matrix calls to test multiple Kubernetes versions simultaneously.
+
+#### Expected Outcomes:
+
+Successful migration of Kubeflow components.
+Hard multi-tenancy for the object storage
+Updated dependencies with resolved CVEs.
+Enhanced CI/CD pipeline supporting multiple Kubernetes versions.
+
+#### Skills Required/Preferred:
+
+GitHub and GitHub Actions
+Containerfile and Kubernetes knowledge
+Experience with Python, Go and JavaScript frameworks
+
+#### Mentors:
+
+@juliusvonkohout
+@thesuperzapper are you interested?
+
+#### Component: Kubeflow
+
+#### Difficulty: Medium
+
+#### Expected Size of the Project: 350 hours
+
+### Project 2
+### Project Title: Maintain Kserve Models Web Application such that we have a proper UI for model serving.
+
+#### Detailed Description:
+
+- Clean up and merge 20 issues and 5 PRs.
+- Implement a better CI/CD pipeline.
+- Potentially migrate the application to /kubeflow/kserve-model-ui.
+- Add features for editing, regression testing, and monitoring/metrics support.
+- Synchronize with kserve 0.14+ changes.
+
+#### Expected Outcomes:
+
+- Resolved issues and merged PRs.
+less CVEs
+- Improved CI/CD pipeline.
+- New editing and regression testing features.
+- Enhanced monitoring and metrics support.
+- Application synchronized with kserve 0.14+ changes.
+
+#### Skills Required/Preferred:
+
+- GitHub Actions
+- Containerfiles
+- JavaScript frameworks
+
+#### Possible Mentors:
+
+Julius von Kohout
+@varodrig do you want to help here?
+Component: Kserve UI
+
+#### Difficulty: Medium
+
+Expected Size of the Project: 175+ hours
+
+
+### Project 3
+### Project Title: Make our service mesh rootless by default and provide secure model inference by default.
+
+#### Detailed Description:
+Secure our service mesh with istio-cni by default kubeflow/manifests#2907 and an option for istio-ambient mesh kubeflow/manifests#2676 with zero overhead namespaces (single waypoint proxy deployment). Furthermore we need to enable a secure Kserve by default kubeflow/manifests#2811. A general plan is also available at
+kubeflow/manifests#2528
+
```

Comment:
- Controllers to create HTTPRoute and AuthorizationPolicies, that align with way-point proxies
- Manifests to also have a flavour of HTTPRoute and updated AuthorizationPolicies